isis/src/base/objs/CSMCamera/CSMCamera.cpp

```diff
@@ -651,18 +651,23 @@ namespace Isis {
 
   /**
    * Set the Target object for the camera model.
+   * 
+   * If no radii can be found in the label from the naif keywords
+   * use the radii from the csm model
    *
    * @param label The label containing information to create the Target from
    */
   void CSMCamera::setTarget(Pvl label) {
     Target *target = new Target(label);
 
-    // get radii from CSM
-    csm::Ellipsoid targetEllipsoid = csm::SettableEllipsoid::getEllipsoid(m_model);
-    std::vector<Distance> radii  = {Distance(targetEllipsoid.getSemiMajorRadius(), Distance::Meters),
-                                    Distance(targetEllipsoid.getSemiMajorRadius(), Distance::Meters),
-                                    Distance(targetEllipsoid.getSemiMinorRadius(), Distance::Meters)};
-    target->setRadii(radii);
+    if (target->radii().size() == 0) {
+      // get radii from CSM
+      csm::Ellipsoid targetEllipsoid = csm::SettableEllipsoid::getEllipsoid(m_model);
+      std::vector<Distance> radii = {Distance(targetEllipsoid.getSemiMajorRadius(), Distance::Meters),
```

Comment:
Is this the same as: if no camera model, default to ellipsoid?

Reply:
Somewhat, the difference here is in the Target changes where we try to read the radii from the naifwords first

Reply:
how often are those different?

Reply:
The isds generated by ALE produce the same semi-major and semi-minor radii but don't capture a third radii if present. For our more common targets (Moon and Mars), the radii are often just a single radii, or a bi-axial ellipsoid. For less common targets, specifically small bodies, we often use a tri-axial radii and that third radii would not be captured by the CSM model